### ISSUE TITLE
If at least 1 node is set to hidden, useNodesInitialized will never return true

### ISSUE THREAD

**Issue by ineentho** [wbkd/react-flow#2953]
### Describe the Bug

`useNodesInitialized` will never return true if at least one node is hidden. Technically this behaviour is what the documentation describes, as no size will be calculated for the hidden node, but I think it would be more useful if `useNodesInitialized` only concidered visible nodes.

My use case is that I want to do automatic layout calculations, but simply skip the nodes that are not visible.

### Your Example Website or App

https://codesandbox.io/s/romantic-jones-wfwu0x?file=/App.js

### Steps to Reproduce the Bug or Issue

1. Open the Codesandbox
2. Check the console, it will log `{nodesInitialized: false}`. 
3. Remove the hidden attribute from the first node
4. Observe that `{nodesInitialized: true}` is logged as expected

### Expected behavior

`useNodesInitialized` should only concider visible nodes

### Platform

Version: 11.6.1

**Comment by moklick** [comment]
I agree here. I thought that hidden nodes would not be taken into account here. In the next release, hidden nodes will be ignored by default and we added an option to configure it https://github.com/wbkd/react-flow/blob/next-release/packages/core/src/hooks/useNodesInitialized.ts#L20-L22

**Comment by moklick** [comment]
available/changed in v11.7.0

**Comment by Varun-Sangai** [comment]
> I agree here. I thought that hidden nodes would not be taken into account here. In the next release, hidden nodes will be ignored by default and we added an option to configure it https://github.com/wbkd/react-flow/blob/next-release/packages/core/src/hooks/useNodesInitialized.ts#L20-L22



> I agree here. I thought that hidden nodes would not be taken into account here. In the next release, hidden nodes will be ignored by default and we added an option to configure it https://github.com/wbkd/react-flow/blob/next-release/packages/core/src/hooks/useNodesInitialized.ts#L20-L22

But useNodesInitialized still giving false  with includeHiddenNodes set to true and atleast one node set to hidden:true
